src/main/java/teammates/ui/webapi/action/GetFeedbackSessionsAction.java

```diff
@@ -20,12 +23,27 @@ protected AuthType getMinAuthLevel() {
 
     @Override
     public void checkSpecificAccessControl() {
+        if (!(userInfo.isStudent || userInfo.isInstructor)) {
+            throw new UnauthorizedAccessException("Student or instructor privilege is required to access this resource.");
+        }
+
         String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
-        if (courseId == null) {
-            gateKeeper.verifyInstructorPrivileges(logic.getAccount(userInfo.getId()));
+        boolean isRequestAsStudent = getRequestParamValue(Const.ParamsNames.IS_IN_RECYCLE_BIN) == null;
+
+        if (isRequestAsStudent) {
+            if (courseId == null) {
+                gateKeeper.verifyStudentPrivileges(logic.getAccount(userInfo.getId()));
```

Comment:
I think the correct way is to assert userInfo.isStudent to be true.

If you take a look at `verifyStudentPrivileges`, it actually do `verifyLoggedInUserPrivileges` which is not correct.